pkg/plugins/policies/core/rules/subsetutils/subset.go

```diff
@@ -337,3 +350,48 @@ func (c *SubsetIter) simplified() Subset {
 
 	return result
 }
+
+// Deduplicate returns a new slice of subsetutils.Subset with duplicates removed.
+func Deduplicate(subsets []Subset) []Subset {
+	seen := make(map[string]struct{})
+	result := make([]Subset, 0, len(subsets))
+
+	for _, s := range subsets {
+		key := canonicalSubset(s)
+		// sort values by key, value, not
+		s.Sorted()
+		if _, exists := seen[key]; !exists {
+			seen[key] = struct{}{}
+			result = append(result, s)
+		}
+	}
+	sort.SliceStable(result, func(i, j int) bool {
```

Comment:
this sort is not really needed, as if we received an already somehow ordered `subsets` as input then by deduplication we won't unstabilize the order